apm-agent-core/src/test/java/co/elastic/apm/agent/AbstractServletTest.java

```diff
@@ -41,15 +49,30 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public abstract class AbstractServletTest {
-    protected static final MockReporter reporter = new MockReporter();
+    protected static MockReporter reporter;
+    protected static ConfigurationRegistry config;
+    protected static ElasticApmTracer tracer;
+
     @Nullable
     private static Server server;
     protected OkHttpClient httpClient;
 
+    @BeforeAll
+    static void setup() {
+        reporter = new MockReporter();
+        config = SpyConfiguration.createSpyConfig();
+        tracer = new ElasticApmTracerBuilder()
```

Comment:
this will conflict with the changes I did in the JDBC PR, thus we need to not forget about this to be sure the right thing is merged afterwards.